lang/rust/avro/src/ser.rs

```diff
@@ -17,9 +17,24 @@
 
 //! Logic for serde-compatible serialization.
 use crate::{types::Value, Error};
+use ref_thread_local::{ref_thread_local, RefThreadLocal};
 use serde::{ser, Serialize};
 use std::{collections::HashMap, iter::once};
 
+ref_thread_local! {
+    /// A thread local that is used to decide how to serialize
+    /// a byte array into Avro `types::Value`.
+    ///
+    /// Depends on the fact that serde's serialization process is single-threaded!
```

Comment:
How about adding this as field of the Serializer struct instead then to avoid depending on this (and because it would probably be cleaner)?
I think that would be the idiomatic way to do it if it's possible.

Reply:
Ah I see why this isn't possible now ^^'

Hmm I've added a comment at https://issues.apache.org/jira/browse/AVRO-3631 that suggests an alternate implementation.